hardware_deps/device.c

```diff
@@ -0,0 +1,21 @@
+#include "device.h"
+#include <stdio.h>
+#include <stdlib.h>
+
+device_t *hardware_deps_init(void) {
+  device_t *dv = (device_t *)malloc(sizeof(device_t));
+#ifdef SIERRA_WP7702
+  dv->get_key = get_key;
+  dv->get_hardware_identifier = get_hardware_identifier;
+  return dv;
+#else
+  fprintf(stderr, "Hardware not implement\n");
+  free(dv);
+  return NULL;
+#endif
+}
+
+retcode_t release_device(device_t *d) {
+  free(d);
+  return RET_OK;
```

Comment:
Do you think we should assign NULL to the pointer of  `*d`?
And it may be good to use `void` instead of `retcode_t`

Reply:
Is the double free problem still work in 2020?

Reply:
I am free for decision. Don't worry